Add ingester.autoforget-unhealthy-timeout opt-in feature (#3919)
grafana/loki: docs/sources/configuration/_index.md

# CLI flag: -ingester.query-store-max-look-back-period
[query_store_max_look_back_period: <duration> | default = 0]

> # Forget ingesters having hearbeat timestamps older than `ring.kvstore.heartbeat_timeout`
> # It is equivalent to clicking on `/ring` `forget` button in the UI: the ingester is removed from the ring
> # It is useful when you are pretty sure that unhealthy nodes won't come back (ex: not using stateful sets or equivalent)
> # You may use `memberlist.rejoin_interval` > 0 to handle network partition cases when using memberlist
> # CLI flag: -ingester.autoforget-unhealthy

Reviewer: I'd like to see the wording (and spelling) improved in this PR.  I've only looked at the docs, and not the implementation.

```suggestion
# Forget about ingesters having heartbeat timestamps older than `ring.kvstore.heartbeat_timeout`.
# This is equivalent to clicking on `/ring` `forget` button in the UI: the ingester is removed from the ring.
# A useful setting when you are sure that an unhealthy node won't return. An example is when not
# using stateful sets or the equivalent.
# You may use `memberlist.rejoin_interval` > 0 to handle network partition cases when using a memberlist.
# CLI flag: -ingester.autoforget-unhealthy
```
